src/rules/iddataweb-verification-workflow.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * @title ID DataWeb Verification Workflow
+ * @overview Verify your user's identity in 180+ countries with ID DataWeb's adaptive Verification Workflows.
+ * @gallery true
+ * @category marketplace
+ *
+ * This configuration allows you to add an ID DataWeb verification workflow to the user’s first time login experience. This will allow your organization to ensure the user is really who they claim to be, aligning with KYC/AML requirements, and mitigating fraud attempts.
+ *
+ */
+
+function iddatawebVerificationWorkflow(user, context, callback) {
+  const { Auth0RedirectRuleUtilities } = require("@auth0/rule-utilities@0.1.0");
+
+  const ruleUtils = new Auth0RedirectRuleUtilities(
+    user,
+    context,
+    configuration
+  );
+
+  const {
+    IDDATAWEB_BASE_URL,
+    IDDATAWEB_CLIENT_ID,
+    IDDATAWEB_CLIENT_SECRET,
+    IDDATAWEB_LOG_JWT,
+    IDDATAWEB_ALWAYS_VERIFY,
+    IDDATAWEB_PREFILL_ATTRIBUTES,
+  } = configuration;
+
+  const idwTokenEndpoint = `${IDDATAWEB_BASE_URL}/axn/oauth2/token`;
+  const idwAuthorizeEndpoint = `${IDDATAWEB_BASE_URL}/axn/oauth2/authorize`;
+  const auth0ContinueUrl = `https://${context.request.hostname}/continue`;
+
+  // initialize app metadata
+  user.app_metadata = user.app_metadata || {};
+  user.app_metadata.iddataweb = user.app_metadata.iddataweb || {};
+
+  // if the user is already verified and we don't need to check, exit
+  if (
+    user.app_metadata.iddataweb.verificationResult === "verified" &&
+    IDDATAWEB_ALWAYS_VERIFY === "off"
+  ) {
+    console.log("user " + user.user_id + " has been previously verified.");
+    return callback(null, user, context);
+  }
+
+  // if coming back from redirect - get token, make policy decision, and update user metadata.
+  if (ruleUtils.isRedirectCallback) {
+    console.log("code from IDW: " + ruleUtils.queryParams.code);
+
+    let options = {
+      method: "POST",
+      url: idwTokenEndpoint,
+      headers: {
+        "Cache-Control": "no-cache",
+        Authorization:
+          "Basic " +
+          Buffer.from(
+            IDDATAWEB_CLIENT_ID + ":" + IDDATAWEB_CLIENT_SECRET
+          ).toString("base64"),
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+      form: {
+        grant_type: "authorization_code",
+        code: ruleUtils.queryParams.code,
+        redirect_uri: auth0ContinueUrl,
+      },
+    };
+
+    request(options, function (error, response, body) {
+      if (error) {
+        return callback(err);
+      }
+
+      let decodedToken;
+      try {
+        decodedToken = jwt.decode(JSON.parse(body).id_token);
+      } catch (tokenError) {
+        return callback(tokenError);
+      }
+
+      console.log("policy decision: " + decodedToken.policyDecision);
+      console.log("score: " + decodedToken.idwTrustScore);
+
+      if (IDDATAWEB_LOG_JWT === "on") {
+        console.log(JSON.stringify(decodedToken));
```

Comment:
Is there a possibility of anything sensitive being in here?

Reply:
Yes - there could be sensitive information stored here if "on". This is sometimes useful for debugging, but shouldn't be used in production mode. Alternatively - we can remove the ability to print the full JWT, and just log additional details only (like transaction ID, etc.) Thoughts?

Reply:
Maybe rename the config key to `IDDATAWEB_DEBUG` to make it more clear that it's not for regular use. You might be able to use it elsewhere for the same purpose.

Reply:
Moved this up a few lines to log before checking ... still would like to change that config key name